src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
+                            else if (string.Compare(name, "user-agent", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                _httpRequestMessage.Headers.UserAgent.Add(ProductInfoHeaderValue.Parse(value));
+                            }
+                            else if (string.Compare(name, "if-modified-since", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                DateTimeOffset modifiedSinceDate;
+                                if (DateTimeOffset.TryParse(value, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowWhiteSpaces | DateTimeStyles.AssumeLocal, out modifiedSinceDate))
+                                {
+                                    _httpRequestMessage.Headers.IfModifiedSince = modifiedSinceDate;
+                                }
+                                else
+                                {
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                                        new ProtocolException(SR.Format(SR.HttpIfModifiedSinceParseError, value)));
+                                }
+                            }
+                            else if (string.Compare(name, "date", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // this will be taken care of by System.Net when we make the request
+                            }
+                            else if (string.Compare(name, "proxy-connection", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                throw ExceptionHelper.PlatformNotSupported("proxy-connection");
+                            }
+                            else if (string.Compare(name, "range", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // we don't support ranges in v1.
```

Basically my question is, do you think we should throw an exception or continue to silently drop it like we do on the desktop clr?